src/components/actions/getApplicationData.ts

```diff
@@ -23,28 +23,26 @@ export const getApplicationData = async (input: {
       trigger_payload?.record_id as number
     ))
 
-  const applicationResult = await DBConnect.getApplicationData(applicationId)
-  if (!applicationResult) throw new Error("Can't get application data")
-
-  const applicationData = applicationResult ? applicationResult : { applicationId }
+  const applicationData: BaseApplicationData = await DBConnect.getApplicationData(applicationId)
+  if (!applicationData) throw new Error("Can't get application data")
 
   const {
-    user: {
-      firstName,
-      lastName,
-      organisation: orgName,
-      username,
-      dateOfBirth,
-      email,
-      permissionNames,
-    },
+    user: { firstName, lastName, organisation, username, dateOfBirth, email, permissionNames },
   } = await getUserInfo({
     userId: applicationData.userId,
-    username: applicationData.username,
-    orgId: applicationData.orgId,
+    orgId: applicationData?.orgId || undefined,
   })
 
-  const userData = { firstName, lastName, orgName, username, dateOfBirth, email, permissionNames }
+  const userData = {
+    firstName,
+    lastName,
+    orgName: organisation?.orgName || null,
+    orgId: organisation?.orgId || null,
+    username,
+    dateOfBirth,
+    email,
+    permissionNames,
+  }
```

Comment:
This is the actual bug fix here -- which would never have happened had the other Typescript definitions been in place :)